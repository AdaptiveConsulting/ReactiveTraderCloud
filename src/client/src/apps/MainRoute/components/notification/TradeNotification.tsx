import moment from 'moment'
import React, { FC } from 'react'
import { Direction, Trade, TradeStatus } from 'rt-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShare } from '@fortawesome/free-solid-svg-icons'
import { library } from '@fortawesome/fontawesome-svg-core'

import {
  Bottom,
  CloseContainer,
  Close,
  MetaContainer,
  MetaTitle,
  Notification,
  Status,
  Top,
  Traded,
} from './styled'
library.add(faShare)
export interface Props {
  trade: Trade
  dismissNotification: () => void
  highlightTradeInBlotter?: () => void
}

const TradeNotification: FC<Props> = ({
  trade: trade,
  dismissNotification,
  highlightTradeInBlotter,
}) => {
  const formattedValueDate = moment(trade.valueDate)
    .format('DD MMM')
    .toUpperCase()

  const isDone = trade.status === TradeStatus.Done

  const tradeStatus = isDone ? '' : 'REJECTED'
  const direction = trade.direction === Direction.Buy ? 'Bought' : 'Sold'
  return (
    <Notification onClick={highlightTradeInBlotter}>
      <Top justifyContent="space-between">
        <Traded isDone={isDone}>
          <div>{`${direction.toUpperCase()} ${trade.dealtCurrency} ${trade.notional}`}</div>
          <div>{`vs ${trade.termsCurrency}`}</div>
        </Traded>

        <Status isDone={isDone}>{tradeStatus.toUpperCase()}</Status>
      </Top>
      <Bottom justifyContent="space-between">
        <MetaContainer>
          <MetaTitle>Rate</MetaTitle>
          <div>{trade.spotRate}</div>
        </MetaContainer>
        <MetaContainer>
          <MetaTitle>Date</MetaTitle>
          <div>SP. {formattedValueDate.toString()}</div>
        </MetaContainer>
        <MetaContainer>
          <MetaTitle>Trade Id</MetaTitle>
          <div>{trade.tradeId}</div>
        </MetaContainer>
        <CloseContainer>
          <Close onClick={dismissNotification}>
            <FontAwesomeIcon icon="share" size="2x" />
          </Close>
        </CloseContainer>
      </Bottom>
    </Notification>
  )
}

export default TradeNotification
