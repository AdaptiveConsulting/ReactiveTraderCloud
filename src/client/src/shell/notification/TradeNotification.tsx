import moment from 'moment'
import React from 'react'
import { Direction, Trade, TradeStatus } from 'rt-types'
import { Bottom, Close, CloseContainer, MetaContainer, MetaTitle, Notification, Status, Top, Traded } from './styled'

export interface Props {
  message: Trade
  dismissNotification: () => void
}

const TradeNotification: React.SFC<Props> = ({ message: trade, dismissNotification }) => {
  const formattedValueDate = trade
    ? moment(trade.valueDate)
        .format('DD MMM')
        .toUpperCase()
    : ''

  const isDone = trade.status === TradeStatus.Done

  const tradeStatus = isDone ? '' : 'REJECTED'
  const direction = trade.direction === Direction.Buy ? 'Bought' : 'Sold'

  return (
    <Notification>
      <Top justifyContent="space-between">
        <Traded isDone={isDone}>
          <div>{`${direction.toUpperCase()} ${trade.dealtCurrency} ${trade.notional}`}</div>
          <div>{`vs ${trade.termsCurrency}`}</div>
        </Traded>

        <Status isDone={isDone}>{tradeStatus.toUpperCase()}</Status>
      </Top>
      <Bottom justifyContent="space-between">
        <MetaContainer>
          <MetaTitle>Rate</MetaTitle>
          <div>{trade.spotRate}</div>
        </MetaContainer>
        <MetaContainer>
          <MetaTitle>Date</MetaTitle>
          <div>SP. {formattedValueDate.toString()}</div>
        </MetaContainer>
        <MetaContainer>
          <MetaTitle>Trade Id</MetaTitle>
          <div>{trade.tradeId}</div>
        </MetaContainer>
        <CloseContainer>
          <Close>
            <i className="fas fa-share fa-lg" onClick={dismissNotification} />
          </Close>
        </CloseContainer>
      </Bottom>
    </Notification>
  )
}

export default TradeNotification
