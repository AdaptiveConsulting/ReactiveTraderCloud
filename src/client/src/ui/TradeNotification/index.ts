export { default } from './tradeNotificationContainer'
export { default as TradeNotification } from './tradeNotification'
export { buildNotification } from './notificationUtils'
