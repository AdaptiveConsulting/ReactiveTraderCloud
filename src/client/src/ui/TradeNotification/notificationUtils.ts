import { timeFormat } from 'd3-time-format'
import { Notification, NotificationType, Trade } from './'

export function buildNotification(trade: Trade = null, error): Notification {
  if (error || !trade) {
    return { message: error, hasError: true, notificationType: NotificationType.Text, trade: null }
  }
  return {
    trade,
    notificationType: NotificationType.Trade,
    hasError: false
  }
}
