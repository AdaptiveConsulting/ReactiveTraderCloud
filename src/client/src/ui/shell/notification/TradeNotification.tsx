import classnames from 'classnames'
import { Direction, Trade, TradeStatus } from 'common/types'
import React from 'react'
import '../../styles/css/index.css'

export interface TradeNotificationProps {
  message: Trade
  dismissNotification: () => void
}

const TradeNotification: React.SFC<TradeNotificationProps> = ({ message: trade, dismissNotification }) => {
  const { statusClassName, tradeSummaryClasses } = getClassNames(trade)
  const formattedValueDate = trade ? trade.valueDate : ''
  const tradeStatus = trade.status === TradeStatus.Done ? trade.status : 'REJECTED'
  const direction = trade.direction === Direction.Buy ? 'Bought' : 'Sold'

  return (
    <div className="notification">
      <div className="notification__content">
        <span className={statusClassName}>{tradeStatus}</span>
        <ul className={tradeSummaryClasses}>
          <li className="notification__summary-item notification__summary-item--direction">{direction}</li>
          <li className="notification__summary-item notification__summary-item--notional">
            {trade.dealtCurrency} {trade.notional}
          </li>
          <li className="notification__summary-item notification__summary-item--currency">vs {trade.termsCurrency}</li>
        </ul>
        <div className="notification__details-items-container">
          <ul className="notification__details-items">
            <li className="notification__details-item notification__details-item--label">Rate</li>
            <li className="notification__details-item notification__details-item--value">{trade.spotRate}</li>
          </ul>
          <ul className="notification__details-items">
            <li className="notification__details-item notification__details-item--label">Date</li>
            <li className="notification__details-item notification__details-item--value">SP. {formattedValueDate}</li>
          </ul>
          <ul className="notification__details-items">
            <li className="notification__details-item notification__details-item--label">Trade Id</li>
            <li className="notification__details-item notification__details-item--value">{trade.tradeId}</li>
          </ul>
        </div>
        <a href="#" className="notification__button--dismiss" onClick={dismissNotification}>
          <i className="notification__button--dismiss-icon fa fa-share" />
        </a>
      </div>
    </div>
  )
}

export default TradeNotification

function getClassNames(trade: Trade) {
  const statusClassName = classnames('notification__status', {
    'notification__status--done': trade.status === TradeStatus.Done,
    'notification__status--rejected': trade.status !== TradeStatus.Done
  })
  const tradeSummaryClasses = classnames('notification__summary-items', {
    'notification__summary-items--rejected': trade.status === TradeStatus.Rejected
  })
  return { statusClassName, tradeSummaryClasses }
}
