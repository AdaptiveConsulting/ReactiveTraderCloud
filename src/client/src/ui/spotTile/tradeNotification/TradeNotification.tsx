import * as React from 'react'
import * as classnames from 'classnames'
import './trade-notification.scss'

interface TradeNotificationProps {
  className: string,
  notification: {
    hasError: boolean
    status: {
      name: string,
    }
    dealtCurrency: string
    notional: number
    termsCurrency: any
    direction: string
    spotRate: number
    formattedValueDate: string
    tradeId: string,
  },
  onDismissedClicked: () => void
}

const tradeStatus = {
  rejected: { name: 'Rejected' },
  done: { name: 'Done' },
  pending: { name: 'Pending' },
}

class TradeNotification extends React.Component<TradeNotificationProps, {}>{
  props: TradeNotificationProps

  renderError() {
    const classNames = classnames(
      'trade-notification',
      'trade-notification--error',
      this.props.className,
    )
    return (
    <div className={classNames}>
      {this.props.notification.hasError}.
      The execution status is unknown. Please contact your sales rep.
        <a
          href="#"
          className="trade-notification__button--dismiss"
          onClick={this.props.onDismissedClicked}>
          Done
        </a>
    </div>)
  }

  render() {
    const { notification, onDismissedClicked, className } = this.props

    if (notification.hasError) {
      return this.renderError()
    }

    const classNames = classnames(
      'trade-notification',
      className,
      { 'trade-notification--rejected': notification.status.name === tradeStatus.rejected.name },
    )
    return (
      <div className={classNames}>
        <span className="trade-notification__trade-status">
          {notification.status.name}
          </span>
        <ul className="trade-notification__summary-items">
          <li
            // tslint:disable-next-line:max-line-length
            className="trade-notification__summary-item trade-notification__summary-item--direction">
            {notification.direction}
          </li>
          <li
            className="trade-notification__summary-item trade-notification__summary-item--notional">
            {notification.dealtCurrency} {notification.notional}
          </li>
          <li
            className="trade-notification__summary-item trade-notification__summary-item--currency">
            <span
              className="trade-notification__label--versus">vs </span>
              {notification.termsCurrency}
          </li>
        </ul>
        <div className="trade-notification__details-items-container">
          <ul className="trade-notification__details-items trade-notification__details-items--rate">
            <li
              className="trade-notification__details-item trade-notification__details-item--label">
              Rate
            </li>
            <li
              className="trade-notification__details-item trade-notification__details-item--value">
              {notification.spotRate}
            </li>
          </ul>
          <ul className="trade-notification__details-items trade-notification__details-items--date">
            <li
              className="trade-notification__details-item trade-notification__details-item--label">
              Date
            </li>
            <li
              className="trade-notification__details-item trade-notification__details-item--value">
              {notification.formattedValueDate}
            </li>
          </ul>
          <ul
             // tslint:disable-next-line:max-line-length
            className="trade-notification__details-items trade-notification__details-items--trade-id">
            <li className="trade-notification__details-item trade-notification__details-item--label">
              Trade ID
            </li>
            <li
              className="trade-notification__details-item trade-notification__details-item--value">
              {notification.tradeId}
            </li>
          </ul>
        </div>
        <a
          href="#"
          className="trade-notification__button--dismiss"
          onClick={onDismissedClicked}>
          <i className="trade-notification__button--dismiss-icon fa fa-share" ></i>
        </a>
      </div>
    )
  }
}

export default TradeNotification
