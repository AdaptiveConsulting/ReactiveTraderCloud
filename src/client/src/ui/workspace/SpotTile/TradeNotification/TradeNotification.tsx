import React from 'react';
import classnames from 'classnames';
import './trade-notification.scss';


interface TradeNotificationProps {
  className: string
  tradeExecutionNotification: {},
  onDismissedClicked: () => void
}

type TradeStatus = {
  rejected: string,
  done: string,
  pending: string
}

const tradeStatus: TradeStatus = {
  rejected: 'Rejected',
  done: 'Done',
  pending: 'Pending'
}

class TradeNotification extends React.Component<TradeNotificationProps, {}>{
  props: TradeNotificationProps

  renderError() {
    const classNames = classnames(
      'trade-notification',
      'trade-notification--error',
      this.props.className
    );
    return (<div className={classNames}>
      {this.props.tradeExecutionNotification.hasError}. The execution status is unknown. Please contact your sales rep.
        <a href='#' className='trade-notification__button--dismiss' onClick={this.props.onDismissedClicked}>Done</a>
    </div>);
  }

  render() {
    const { tradeExecutionNotification, onDismissedClicked, className } = this.props;

    if (tradeExecutionNotification.hasError) {
      return this.renderError()
    }

    const classNames = classnames(
      'trade-notification',
      className,
      { 'trade-notification--rejected': tradeExecutionNotification.status === tradeStatus.rejected }
    )
    return (
      <div className={classNames}>
        <span className="trade-notification__trade-status">{tradeExecutionNotification.status.name}</span>
        <ul className="trade-notification__summary-items">
          <li className="trade-notification__summary-item trade-notification__summary-item--direction">
            {tradeExecutionNotification.direction}
          </li>
          <li className="trade-notification__summary-item trade-notification__summary-item--notional">
            {tradeExecutionNotification.dealtCurrency} {tradeExecutionNotification.notional}
          </li>
          <li className="trade-notification__summary-item trade-notification__summary-item--currency">
            <span className="trade-notification__label--versus">vs </span>{tradeExecutionNotification.termsCurrency}
          </li>
        </ul>
        <div className="trade-notification__details-items-container">
          <ul className="trade-notification__details-items trade-notification__details-items--rate">
            <li className="trade-notification__details-item trade-notification__details-item--label">
              Rate
            </li>
            <li className="trade-notification__details-item trade-notification__details-item--value">
              {tradeExecutionNotification.spotRate}
            </li>
          </ul>
          <ul className="trade-notification__details-items trade-notification__details-items--date">
            <li className="trade-notification__details-item trade-notification__details-item--label">
              Date
            </li>
            <li className="trade-notification__details-item trade-notification__details-item--value">
              {tradeExecutionNotification.formattedValueDate}
            </li>
          </ul>
          <ul className="trade-notification__details-items trade-notification__details-items--trade-id">
            <li className="trade-notification__details-item trade-notification__details-item--label">
              Trade ID
            </li>
            <li className="trade-notification__details-item trade-notification__details-item--value">
              {tradeExecutionNotification.tradeId}
            </li>
          </ul>
        </div>
        <a
          href="#"
          className="trade-notification__button--dismiss"
          onClick={onDismissedClicked}>
          <i className="trade-notification__button--dismiss-icon fa fa-share" ></i>
        </a>
      </div>
    )
  }
}

export default TradeNotification
