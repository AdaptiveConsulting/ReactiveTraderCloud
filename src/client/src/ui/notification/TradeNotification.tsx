import * as React from 'react'
import './notification.scss'
import * as classnames from 'classnames'
import Direction from '../../services/model/direction'
import TradeStatus from '../../services/model/tradeStatus'
import '../common/styles/_base.scss'
import '../common/styles/_fonts.scss'
const DONE = 'Done'
const REJECTED = 'Rejected'
const BOUGHT = 'Bought'
const SOLD = 'Sold'

export interface TradeNotificationProps {
  message: any
  dismissNotification: any
}

export default class TradeNotification extends React.Component<TradeNotificationProps, {}> {
  render() {
    const trade = this.props.message
    const statusClassName = classnames(
      'notification__status',
      {
        'notification__status--done': trade.tradeStatus === DONE,
        'notification__status--rejected' : trade.tradeStatus !== DONE,
      },
    )
    const tradeSummaryClasses = classnames(
      'notification__summary-items',
      {
        'notification__summary-items--rejected': trade.tradeStatus === REJECTED,
      },
    )
    const tradeStatus = trade.tradeStatus === TradeStatus.Done ? trade.tradeStatus : REJECTED.toUpperCase()
    const direction = trade.direction === Direction.Buy.name ? BOUGHT : SOLD

    return (
      <div className="notification">
        <div className="notification__content">
          <span className={statusClassName}>{tradeStatus}</span>
          <ul className={tradeSummaryClasses}>
            <li className="notification__summary-item notification__summary-item--direction">{direction}</li>
            <li className="notification__summary-item notification__summary-item--notional">{trade.baseCurrency} {trade.notional}</li>
            <li className="notification__summary-item notification__summary-item--currency">vs {trade.termsCurrency}</li>
          </ul>
          <div className="notification__details-items-container">
            <ul className="notification__details-items">
              <li className="notification__details-item notification__details-item--label">Rate</li>
              <li className="notification__details-item notification__details-item--value">{trade.spotRate}</li>
            </ul>
            <ul className="notification__details-items">
              <li className="notification__details-item notification__details-item--label">Date</li>
              <li className="notification__details-item notification__details-item--value">{trade.valueDate}</li>
            </ul>
            <ul className="notification__details-items">
              <li className="notification__details-item notification__details-item--label">Trade Id</li>
              <li className="notification__details-item notification__details-item--value">{trade.tradeId}</li>
            </ul>
          </div>
          <a href="#" className="notification__button--dismiss" onClick={() => this.props.dismissNotification()}><i className="notification__button--dismiss-icon fa fa-share" ></i></a>
        </div>
      </div>
    )
  }
}
