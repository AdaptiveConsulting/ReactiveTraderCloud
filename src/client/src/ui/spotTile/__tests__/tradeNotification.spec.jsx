import * as React from 'react'
import TradeNotification from '../TradeNotification'
import ShallowRenderer from 'react-test-renderer/shallow'
import { mount } from 'enzyme'

describe.only('TradeNotification', () => {

  test('dismisses the error message when hasError', () => {
    const mockOnDismissedClicked = jest.fn(() => {})
    const notification = {
      hasError: true,
      status: {
        name: 'Rejected',
      },
      dealtCurrency: 'GBP',
      notional: 12345,
      termsCurrency: 'GBP',
      direction: 'Up',
      spotRate: 1,
      formattedValueDate: '12/12/2012',
      tradeId: 'ABC',
    }
    const component = <TradeNotification
          className="spot-tile__trade-summary"
          notification={notification}
          onDismissedClicked={mockOnDismissedClicked}/>

    const mounted = mount(component)
    mounted.find('a').simulate('click')
    expect(mockOnDismissedClicked).toBeCalled()
  })

  test('renders correct markup for all statuses', () => {
    const tradeStatus = [
      { name: 'Rejected' },
      { name: 'Done' },
      { name: 'Pending' },
    ]

    tradeStatus.forEach((status) => {
      const notification = {
        hasError: false,
        status: status,
        dealtCurrency: 'GBP',
        notional: 12345,
        termsCurrency: 'GBP',
        direction: 'Up',
        spotRate: 1,
        formattedValueDate: '12/12/2012',
        tradeId: 'ABC',
      }
      const component = <TradeNotification
            className="spot-tile__trade-summary"
            notification={notification}
            onDismissedClicked={() => {}}/>

      const shallowRenderer = new ShallowRenderer()
      shallowRenderer.render(component)
      expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
    })

  })
})
