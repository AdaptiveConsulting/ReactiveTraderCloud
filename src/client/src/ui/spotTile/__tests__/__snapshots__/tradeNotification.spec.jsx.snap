// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TradeNotification renders correct markup for all statuses 1`] = `
<div
  className="trade-notification spot-tile__trade-summary trade-notification--rejected"
>
  <span
    className="trade-notification__trade-status"
  >
    Rejected
  </span>
  <ul
    className="trade-notification__summary-items"
  >
    <li
      className="trade-notification__summary-item trade-notification__summary-item--direction"
    >
      Up
    </li>
    <li
      className="trade-notification__summary-item trade-notification__summary-item--notional"
    >
      GBP
       
      12345
    </li>
    <li
      className="trade-notification__summary-item trade-notification__summary-item--currency"
    >
      <span
        className="trade-notification__label--versus"
      >
        vs 
      </span>
      GBP
    </li>
  </ul>
  <div
    className="trade-notification__details-items-container"
  >
    <ul
      className="trade-notification__details-items trade-notification__details-items--rate"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Rate
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        1
      </li>
    </ul>
    <ul
      className="trade-notification__details-items trade-notification__details-items--date"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Date
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        12/12/2012
      </li>
    </ul>
    <ul
      className="trade-notification__details-items trade-notification__details-items--trade-id"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Trade ID
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        ABC
      </li>
    </ul>
  </div>
  <a
    className="trade-notification__button--dismiss"
    href="#"
    onClick={[Function]}
  >
    <i
      className="trade-notification__button--dismiss-icon fa fa-share"
    />
  </a>
</div>
`;

exports[`TradeNotification renders correct markup for all statuses 2`] = `
<div
  className="trade-notification spot-tile__trade-summary"
>
  <span
    className="trade-notification__trade-status"
  >
    Done
  </span>
  <ul
    className="trade-notification__summary-items"
  >
    <li
      className="trade-notification__summary-item trade-notification__summary-item--direction"
    >
      Up
    </li>
    <li
      className="trade-notification__summary-item trade-notification__summary-item--notional"
    >
      GBP
       
      12345
    </li>
    <li
      className="trade-notification__summary-item trade-notification__summary-item--currency"
    >
      <span
        className="trade-notification__label--versus"
      >
        vs 
      </span>
      GBP
    </li>
  </ul>
  <div
    className="trade-notification__details-items-container"
  >
    <ul
      className="trade-notification__details-items trade-notification__details-items--rate"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Rate
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        1
      </li>
    </ul>
    <ul
      className="trade-notification__details-items trade-notification__details-items--date"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Date
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        12/12/2012
      </li>
    </ul>
    <ul
      className="trade-notification__details-items trade-notification__details-items--trade-id"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Trade ID
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        ABC
      </li>
    </ul>
  </div>
  <a
    className="trade-notification__button--dismiss"
    href="#"
    onClick={[Function]}
  >
    <i
      className="trade-notification__button--dismiss-icon fa fa-share"
    />
  </a>
</div>
`;

exports[`TradeNotification renders correct markup for all statuses 3`] = `
<div
  className="trade-notification spot-tile__trade-summary"
>
  <span
    className="trade-notification__trade-status"
  >
    Pending
  </span>
  <ul
    className="trade-notification__summary-items"
  >
    <li
      className="trade-notification__summary-item trade-notification__summary-item--direction"
    >
      Up
    </li>
    <li
      className="trade-notification__summary-item trade-notification__summary-item--notional"
    >
      GBP
       
      12345
    </li>
    <li
      className="trade-notification__summary-item trade-notification__summary-item--currency"
    >
      <span
        className="trade-notification__label--versus"
      >
        vs 
      </span>
      GBP
    </li>
  </ul>
  <div
    className="trade-notification__details-items-container"
  >
    <ul
      className="trade-notification__details-items trade-notification__details-items--rate"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Rate
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        1
      </li>
    </ul>
    <ul
      className="trade-notification__details-items trade-notification__details-items--date"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Date
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        12/12/2012
      </li>
    </ul>
    <ul
      className="trade-notification__details-items trade-notification__details-items--trade-id"
    >
      <li
        className="trade-notification__details-item trade-notification__details-item--label"
      >
        Trade ID
      </li>
      <li
        className="trade-notification__details-item trade-notification__details-item--value"
      >
        ABC
      </li>
    </ul>
  </div>
  <a
    className="trade-notification__button--dismiss"
    href="#"
    onClick={[Function]}
  >
    <i
      className="trade-notification__button--dismiss-icon fa fa-share"
    />
  </a>
</div>
`;
